### PR TITLE
Add dropdown component

### PR DESCRIPTION
**Pull request by elmarques**
# What

This PR adds the dropdown component to Astro.

# Why

To allow users to select pre-defined options.

# How

* Add the component styles and markup
* Add the component page to the documentation
* Add sample script for the component to work

# Sample

Captura de Tela 2020-04-08 às 14 53 53

---

Captura de Tela 2020-04-08 às 14 54 53

---

Captura de Tela 2020-04-08 às 14 55 11

# Refs

* [Clubhouse card](https://app.clubhouse.io/magnetis/story/34242/criar-componente-dropdown-para-o-astro-css) [ch34242]

### CODE REVIEW

File: docs/dropdown.mdx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+<Playground style={{ height: '210px' }}>
+  <div id='dropdownValidated' className='a-dropdown a-dropdown--validated'>
+    <input
+      id='dropdownInputValidated'
+      type='text'
+      aria-labelledby='dropdownLabelValidated'
+      placeholder='Dropdown'
+    />
+    <i />
```
Comment (AgtLucas) [inline docs/dropdown.mdx:58]: What's this?

Comment (AgtLucas) [reply docs/dropdown.mdx:58]: I see what is is now. What do you think about using a `span` instead?

I know we are using `i` as icons, but that it's not semantically right.

Comment (elmarques) [reply docs/dropdown.mdx:58]: No problem, I'm fixing the thousands of CSS lint messages 🙀 and I'm already changing on this commit.

File: docs/dropdown.mdx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    <label id='dropdownLabelInvalid' htmlFor='dropdownInputInvalid'>
+      Invalid
+    </label>
+    <nav aria-label='State list'>
+      <a role='option'>First option</a>
+      <a role='option'>Second option</a>
+    </nav>
```
Comment (AgtLucas) [inline docs/dropdown.mdx:81]: Isn't possible to use `<select>` and `<option>` here?

Or something like this? 

https://a11y-guidelines.orange.com/web_EN/exemples/simple-menu/simple-menu.html

File: docs/dropdown.mdx (hunk 1 above)
Comment (AgtLucas) [reply docs/dropdown.mdx:58]: Ahaha cool! :D

Comment (elmarques) [reply docs/dropdown.mdx]: Done c22346c and c32c541.

File: docs/dropdown.mdx (hunk 2 above)
Comment (elmarques) [reply docs/dropdown.mdx]: Changed to ul in c22346c and c32c541.

File: .stylelintrc.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ module.exports = {
     'font-family-name-quotes': 'always-unless-keyword',
     'string-quotes': 'single',
     'rule-empty-line-before': 'always',
-    'order/properties-order': propertyOrder
+    'order/properties-order': propertyOrder,
+    'no-descending-specificity': null
```
Comment (AgtLucas) [inline .stylelintrc.js:11]: <3

File: docs/dropdown.mdx (hunk 1 above)
Comment (AgtLucas) [reply docs/dropdown.mdx]: @elmarques Cool!

File: docs/dropdown.mdx (hunk 2 above)
Comment (AgtLucas) [reply docs/dropdown.mdx]: @elmarques amazing!

**Pull request merged**